--- a/x265/source/CMakeLists.txt
+++ b/x265/source/CMakeLists.txt
@@ -266,7 +266,7 @@
 endif(GCC)
 
 find_package(Yasm)
-if(YASM_FOUND AND X86)
+if(YASM_FOUND AND X86 AND X64)
     if (YASM_VERSION_STRING VERSION_LESS "1.2.0")
         message(STATUS "Yasm version ${YASM_VERSION_STRING} is too old. 1.2.0 or later required")
         option(ENABLE_ASSEMBLY "Enable use of assembly coded primitives" OFF)
@@ -278,11 +278,6 @@
     option(ENABLE_ASSEMBLY "Enable use of assembly coded primitives" OFF)
 endif()
 
-# no need to have slow encoding on x86
-if(X86 AND NOT ENABLE_ASSEMBLY)
-    message(FATAL_ERROR "Yasm 1.2.0 or later must be installed")
-endif()
-
 option(CHECKED_BUILD "Enable run-time sanity checks (debugging)" OFF)
 if(CHECKED_BUILD)
     add_definitions(-DCHECKED_BUILD=1)
