--- a/gdtoa/arithchk.c
+++ b/gdtoa/arithchk.c
@@ -106,10 +106,10 @@
 	long Cray1;
 
 	/* Cray1 = 4617762693716115456 -- without overflow on non-Crays */
-	Cray1 = printf(emptyfmt) < 0 ? 0 : 4617762;
-	if (printf(emptyfmt, Cray1) >= 0)
+	Cray1 = printf("%s", emptyfmt) < 0 ? 0 : 4617762;
+	if (printf("%s%ld", emptyfmt, Cray1) >= 0)
 		Cray1 = 1000000*Cray1 + 693716;
-	if (printf(emptyfmt, Cray1) >= 0)
+	if (printf("%s%ld", emptyfmt, Cray1) >= 0)
 		Cray1 = 1000000*Cray1 + 115456;
 	u.d = 1e13;
 	if (u.L == Cray1)
