--- a/configure
+++ b/configure
@@ -272,9 +272,7 @@
 
 printf "checking for compiler SSE4.1 support..."
 if cxx_check "${CXXFLAGS} -msse4.1 ${LDFLAGS}" ; then
-    if cxx_check "${CXXFLAGS} -march=native -mtune=native ${LDFLAGS}" ; then
-        CXXFLAGS="$CXXFLAGS -march=native -mtune=native"
-    elif cxx_check "${CXXFLAGS} -mpopcnt ${LDFLAGS}" ; then
+    if cxx_check "${CXXFLAGS} -mpopcnt ${LDFLAGS}" ; then
         CXXFLAGS="$CXXFLAGS -msse4.1 -mpopcnt"
     else
         CXXFLAGS="$CXXFLAGS -msse4.1"
@@ -291,6 +289,22 @@
     exit 1 
 fi
 echo "OK"
+
+printf "checking for compiler SSE4.2 support..."
+if cxx_check "${CXXFLAGS} -msse4.2 ${LDFLAGS}" ; then
+    CXXFLAGS="$CXXFLAGS -msse4.2"
+else
+    echo "SSE4.2 not supported with this compiler." 
+    exit 1
+fi
+echo "OK"
+
+printf "checking for system SSE4.2 support..."
+if [ `cat /proc/cpuinfo | grep "sse4_2" | wc --lines` -eq 0 ]; then
+    echo "System does not support SSE4.2."
+    exit 1 
+fi
+echo "OK"
 
 printf "checking for __rdtscp()..."
 if ! cxx_check "${CXXFLAGS} ${LDFLAGS}" "" "x86intrin.h" "unsigned int a = 0; auto i = __rdtscp(&a);" ; then
