--- a/SFML-1.6/src/SFML/Audio/Makefile
+++ b/SFML-1.6/src/SFML/Audio/Makefile
@@ -19,7 +19,7 @@
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJVORBIS)
 
 libsfml-audio.so: $(OBJ) $(OBJVORBIS)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJVORBIS) -lsndfile -lopenal
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJVORBIS) -lsndfile -lopenal $(LIBPATH)/libsfml-system.so.$(VERSION)
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
--- a/SFML-1.6/src/SFML/Graphics/Makefile
+++ b/SFML-1.6/src/SFML/Graphics/Makefile
@@ -27,7 +27,7 @@
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
 
 libsfml-graphics.so: $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB) -lfreetype -lX11 -lGL
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) $(OBJJPEG) $(OBJPNG) $(OBJSOIL) $(OBJZLIB) -lfreetype $(LIBPATH)/libsfml-system.so.$(VERSION) $(LIBPATH)/libsfml-window.so.$(VERSION) -lGL
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS) -I/usr/include/freetype2
--- a/SFML-1.6/src/SFML/Makefile
+++ b/SFML-1.6/src/SFML/Makefile
@@ -37,7 +37,7 @@
 	@(mkdir -p ../../lib)
 	@(cd ./System && $(MAKE))
 
-sfml-window:
+sfml-window: sfml-system
 	@(mkdir -p ../../lib)
 	@(cd ./Window && $(MAKE))
 
@@ -45,7 +45,7 @@
 	@(mkdir -p ../../lib)
 	@(cd ./Network && $(MAKE))
 
-sfml-graphics:
+sfml-graphics: sfml-system sfml-window
 	@(mkdir -p ../../lib)
 	@(cd ./Graphics && $(MAKE))
 
--- a/SFML-1.6/src/SFML/Window/Makefile
+++ b/SFML-1.6/src/SFML/Window/Makefile
@@ -18,7 +18,7 @@
 	$(AR) $(ARFLAGS) $(LIBNAME) $(OBJ)
 
 libsfml-window.so: $(OBJ)
-	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) -lX11 -lXrandr -lGL
+	$(CPP) $(LDFLAGS) -Wl,-soname,$(LIB).$(VERSION) -o $(LIBNAME) $(OBJ) $(OBJGLEW) -lX11 -lXrandr -lGL $(LIBPATH)/libsfml-system.so.$(VERSION)
 
 $(OBJ): %.o: %.cpp
 	$(CPP) -o $@ -c $< $(CFLAGS)
