--- a/programs/dtfile/dterror.src
+++ b/programs/dtfile/dterror.src
@@ -1,4 +1,4 @@
-XCOMM! CDE_INSTALLATION_TOP/bin/dtksh
+XCOMM! /usr/lib/dt/bin/dtksh
 XCOMM $XConsortium: dterror.src /main/4 1996/11/19 11:40:20 drk $
 XCOMM ####################################################################
 XCOMM   Display an error dialog using the argument(s) as the message
--- a/programs/dtfile/dtfile_error.cpp
+++ b/programs/dtfile/dtfile_error.cpp
@@ -1,4 +1,4 @@
-XCOMM! CDE_INSTALLATION_TOP/bin/dtksh
+XCOMM! /usr/lib/dt/bin/dtksh
 XCOMM $XConsortium: dtfile_error.cpp /main/3 1996/04/21 19:29:14 drk $
 XCOMM ##################################################################
 XCOMM  Display an error dialog using the argument(s) as the message
--- a/programs/dthelp/dthelpgen/dthelpgen.dtsh
+++ b/programs/dthelp/dthelpgen/dthelpgen.dtsh
@@ -1,4 +1,4 @@
-#! /usr/dt/bin/dtksh
+#! /usr/lib/dt/bin/dtksh
 #####################################################################
 ###  File:              dthelpgen.dtsh
 ###
@@ -104,9 +104,9 @@ OkButton()
 
 Initialize
 
-if [[ -r /usr/dt/lib/dtksh/DtFuncs.dtsh ]]
+if [[ -r /usr/lib/dt/lib/dtksh/DtFuncs.dtsh ]]
 then
-	. /usr/dt/lib/dtksh/DtFuncs.dtsh
+	. /usr/lib/dt/lib/dtksh/DtFuncs.dtsh
 else
     echo Sorry--cannot find initialization file.
     Exit $NO_INIT_FILE_ERR
--- a/programs/dthelp/dthelpprint/dthelpprint.sh
+++ b/programs/dthelp/dthelpprint/dthelpprint.sh
@@ -62,9 +62,9 @@ usage()
 ### These env vars conform to the dtlp interface
 #########################################################
 # define which executables to use
-ProgDtKsh=/usr/dt/bin/dtksh
-ProgDtAction=/usr/dt/bin/dtaction
-ProgDtLp=/usr/dt/bin/dtlp          # only executed directly if Print action not avail
+ProgDtKsh=/usr/lib/dt/bin/dtksh
+ProgDtAction=/usr/lib/dt/bin/dtaction
+ProgDtLp=/usr/lib/dt/bin/dtlp          # only executed directly if Print action not avail
 ProgLp=/usr/bin/lp
 ActionPrint=Print
 
--- a/programs/dtprintegrate/dtlp.src
+++ b/programs/dtprintegrate/dtlp.src
@@ -1,4 +1,4 @@
-XCOMM! DTKORNSHELL
+XCOMM! /usr/lib/dt/bin/dtksh
 XCOMM $TOG: dtlp.src /main/17 1998/07/31 17:03:31 mgreess $
 XCOMM 
 XCOMM (c) Copyright 1996 Digital Equipment Corporation.
--- a/programs/dtprintegrate/dtlpsetup.src
+++ b/programs/dtprintegrate/dtlpsetup.src
@@ -14,10 +14,10 @@ XCOMM  US Government Users Restricted Rights - Use, duplication or
 XCOMM  disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 XCOMM
 XCOMM
-XCOMM Add /usr/dt/lib/nls/msg/%L/%N.cat to NLSPATH
+XCOMM Add /usr/lib/dt/lib/nls/msg/%L/%N.cat to NLSPATH
 XCOMM
 
-NLSPATH=/usr/dt/lib/nls/msg/%L/%N.cat:$NLSPATH
+NLSPATH=/usr/lib/dt/lib/nls/msg/%L/%N.cat:$NLSPATH
 export NLSPATH
 
 XCOMM Return Codes
@@ -224,7 +224,7 @@ if  [ $NOT_SUPPORTED ]; then
 \t%1$s \n' "$ARGS"
 fi
 
-/usr/dt/bin/dtlp $banner_title $lpdest $print_command $copy_count $other_options $user_filename $a $e $h $r $s $v $w $print_file
+/usr/lib/dt/bin/dtlp $banner_title $lpdest $print_command $copy_count $other_options $user_filename $a $e $h $r $s $v $w $print_file
  
 status=$?
 
--- a/programs/dtprintegrate/dtprintegrate.src
+++ b/programs/dtprintegrate/dtprintegrate.src
@@ -36,7 +36,7 @@ XCOMM ##  Set default values.
 XCOMM ##
 COMMAND_NAME=dtprintegrate
 
-ACTION_STUB_FILE=/usr/dt/appconfig/appmanager/C/Desktop_Apps/Dtcalc
+ACTION_STUB_FILE=/usr/lib/dt/appconfig/appmanager/C/Desktop_Apps/Dtcalc
 
 base_icon_name="Fpprnt"
 
@@ -94,7 +94,7 @@ Initialize()
     HASH
     HASH Search path for printer icons...
     HASH
-    DEFAULT_PATH=.,$HOME/.dt/icons,/usr/dt/appconfig/icons/C
+    DEFAULT_PATH=.,$HOME/.dt/icons,/usr/lib/dt/appconfig/icons/C
     DTPRINTICONPATH=${DTPRINTICONPATH:-$DEFAULT_PATH}
 
     HASH
@@ -229,7 +229,7 @@ MakeDatabaseFile()
             echo "        EXEC_STRING             env LPDEST=$printer_name \\"
 	fi
 
-	echo "				/usr/dt/bin/dtaction Print %(File)Arg_1%"
+	echo "				/usr/lib/dt/bin/dtaction Print %(File)Arg_1%"
 	) >> $databasefile_path
 
 	if (( $help_flag ))
@@ -258,7 +258,7 @@ MakeDatabaseFile()
 	else
 	    echo "	EXEC_STRING		env LPDEST=$printer_name \\"
 	fi
-	echo "				/usr/dt/bin/dtaction Dtqueueinfo"
+	echo "				/usr/lib/dt/bin/dtaction Dtqueueinfo"
 	echo "}"
 	) >> $databasefile_path
 }
@@ -523,27 +523,27 @@ ReloadActions() {
 	HASH
 	HASH  User wants to force reloading actions
 	HASH
-		if [[ -x /usr/dt/bin/dtaction  &&  -n "$DISPLAY" ]]
+		if [[ -x /usr/lib/dt/bin/dtaction  &&  -n "$DISPLAY" ]]
 		then
-			/usr/dt/bin/dtaction ReloadActions
+			/usr/lib/dt/bin/dtaction ReloadActions
 		fi
 	elif [[ "$display_host" = "$session_host" ]]
 	HASH
 	HASH  Our session server is running on the desktop
 	HASH
 	then
-		if [ -x /usr/dt/bin/dtaction ]
+		if [ -x /usr/lib/dt/bin/dtaction ]
 		then
-			/usr/dt/bin/dtaction ReloadActions
+			/usr/lib/dt/bin/dtaction ReloadActions
 		fi
 	elif [[ -z "$display_host" && -n "$DISPLAY" ]]
 	then
 	HASH
 	HASH  Covers the case in which the simple value ":0.0" is our $DISPLAY
 	HASH
-		if [ -x /usr/dt/bin/dtaction ]
+		if [ -x /usr/lib/dt/bin/dtaction ]
 		then
-			/usr/dt/bin/dtaction ReloadActions
+			/usr/lib/dt/bin/dtaction ReloadActions
 		fi
 	fi
 }
