Index: yasm-1.3.0/frontends/tasm/tasm-options.c
===================================================================
--- yasm-1.3.0.orig/frontends/tasm/tasm-options.c
+++ yasm-1.3.0/frontends/tasm/tasm-options.c
@@ -65,15 +66,18 @@ parse_cmdline(int argc, char **argv, opt
 
                     param = &argv[0][1+len];
                     if (options[i].takes_param) {
-                        if (param[0] == '\0') {
+                        if (param[0] != '\0') {
+                            /* do nothing */
+                        } else if (argc > 1) {
+                            param = argv[1];
+                            argc--;
+                            argv++;
+                        } else {
                             print_error(
-                                _("option `-%c' needs an argument!"),
+                                _("option `-%s' needs an argument!"),
                                 options[i].opt);
                             errors++;
                             goto fail;
-                        } else {
-                            argc--;
-                            argv++;
                         }
                     } else
                         param = NULL;
