Index: pidgin-2.10.11/libpurple/protocols/irc/irc.h
===================================================================
--- pidgin-2.10.11.orig/libpurple/protocols/irc/irc.h
+++ pidgin-2.10.11/libpurple/protocols/irc/irc.h
@@ -34,7 +34,7 @@
 #include "roomlist.h"
 #include "sslconn.h"
 
-#define IRC_DEFAULT_SERVER "irc.freenode.net"
+#define IRC_DEFAULT_SERVER "irc.ubuntu.com"
 #define IRC_DEFAULT_PORT 6667
 #define IRC_DEFAULT_SSL_PORT 994
 
