=== modified file 'pidgin/gtkdocklet.c'
--- a/pidgin/gtkdocklet.c
+++ b/pidgin/gtkdocklet.c
@@ -839,7 +839,7 @@
 
 	purple_prefs_add_none(PIDGIN_PREFS_ROOT "/docklet");
 	purple_prefs_add_bool(PIDGIN_PREFS_ROOT "/docklet/blink", FALSE);
-	purple_prefs_add_string(PIDGIN_PREFS_ROOT "/docklet/show", "always");
+	purple_prefs_add_string(PIDGIN_PREFS_ROOT "/docklet/show", "never");
 	purple_prefs_connect_callback(docklet_handle, PIDGIN_PREFS_ROOT "/docklet/show",
 				    docklet_show_pref_changed_cb, NULL);
 
