Index: pidgin-2.10.11/pidgin/gtkprefs.c
===================================================================
--- pidgin-2.10.11.orig/pidgin/gtkprefs.c
+++ pidgin-2.10.11/pidgin/gtkprefs.c
@@ -1339,6 +1339,7 @@ interface_page(void)
 	GtkWidget *label;
 	GtkSizeGroup *sg;
 	GList *names = NULL;
+	GtkWidget *scrolledWindow1024x600;
 
 	ret = gtk_vbox_new(FALSE, PIDGIN_HIG_CAT_SPACE);
 	gtk_container_set_border_width(GTK_CONTAINER(ret), PIDGIN_HIG_BORDER);
@@ -1414,6 +1415,15 @@ interface_page(void)
 	keyboard_shortcuts(ret);
 
 	gtk_widget_show_all(ret);
+
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		ret = scrolledWindow1024x600;
+		gtk_widget_show(ret);
+	}
+
 	g_object_unref(sg);
 	return ret;
 }
@@ -1457,6 +1467,7 @@ conv_page(void)
 	GtkWidget *hbox;
 	GtkWidget *checkbox;
 	GtkWidget *spin_button;
+	GtkWidget *scrolledWindow1024x600;
 
 	ret = gtk_vbox_new(FALSE, PIDGIN_HIG_CAT_SPACE);
 	gtk_container_set_border_width(GTK_CONTAINER(ret), PIDGIN_HIG_BORDER);
@@ -1570,6 +1581,14 @@ conv_page(void)
 					 G_CALLBACK(formatting_clear_cb), NULL);
 	sample_imhtml = imhtml;
 
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		gtk_widget_show(ret);
+		ret = scrolledWindow1024x600;
+	}
+
 	gtk_widget_show(ret);
 
 	return ret;
@@ -1935,6 +1954,7 @@ browser_page(void)
 	GtkWidget *ret, *vbox, *hbox, *label, *entry, *browser_button;
 	GtkSizeGroup *sg;
 	GList *browsers = NULL;
+	GtkWidget *scrolledWindow1024x600;
 
 	ret = gtk_vbox_new(FALSE, PIDGIN_HIG_CAT_SPACE);
 	gtk_container_set_border_width (GTK_CONTAINER (ret), PIDGIN_HIG_BORDER);
@@ -2020,6 +2040,14 @@ browser_page(void)
 		g_object_unref(sg);
 	}
 
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		ret = scrolledWindow1024x600;
+		gtk_widget_show(ret);
+	}
+
 	return ret;
 }
 #endif /*_WIN32*/
@@ -2466,6 +2494,7 @@ sound_page(void)
 	int j;
 	const char *file;
 	char *pref;
+	GtkWidget *scrolledWindow1024x600;
 #ifndef _WIN32
 	GtkWidget *dd;
 	GtkWidget *entry;
@@ -2637,6 +2666,15 @@ sound_page(void)
 	gtk_box_pack_start(GTK_BOX(hbox), button, FALSE, FALSE, 1);
 
 	gtk_widget_show_all(ret);
+
+	if (gdk_screen_get_default() != NULL && gdk_screen_get_height(gdk_screen_get_default()) <= 700) {
+		scrolledWindow1024x600 = gtk_scrolled_window_new(NULL,NULL);
+		gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),GTK_POLICY_NEVER,GTK_POLICY_AUTOMATIC);
+		gtk_scrolled_window_add_with_viewport(GTK_SCROLLED_WINDOW(scrolledWindow1024x600),ret);
+		ret = scrolledWindow1024x600;
+		gtk_widget_show(ret);
+	}
+
 	g_object_unref(sg);
 
 	return ret;
