Origin: https://developer.pidgin.im/attachment/ticket/15575/dtmf-pad.patch
Bug-Upstream: https://developer.pidgin.im/ticket/15575
Description: add dial-pad to the pidgin dialer
--- a/pidgin/gtkmedia.c
+++ b/pidgin/gtkmedia.c
@@ -759,6 +759,171 @@
 }
 
 static void
+phone_dtmf_pressed_cb(GtkButton *button, gpointer user_data)
+{
+	PurpleMediaManager *manager = purple_media_manager_get();
+	GstElement *pipeline;
+
+	pipeline = purple_media_manager_get_pipeline(manager);
+
+	if (pipeline) {
+		GstStructure *structure;
+		GstEvent *event;
+		gint num = GPOINTER_TO_INT(user_data);
+
+		structure = gst_structure_new("dtmf-event",
+						"type", G_TYPE_INT, 1,
+						"number", G_TYPE_INT, num,
+						"volume", G_TYPE_INT, 25,
+						"start", G_TYPE_BOOLEAN, TRUE, NULL);
+
+		event = gst_event_new_custom(GST_EVENT_CUSTOM_UPSTREAM, structure);
+		gst_element_send_event(pipeline, event);
+	}
+}
+
+static void
+phone_dtmf_released_cb(GtkButton *button, gpointer user_data)
+{
+	PurpleMediaManager *manager = purple_media_manager_get();
+	GstElement *pipeline;
+
+	pipeline = purple_media_manager_get_pipeline(manager);
+
+	if (pipeline) {
+		GstStructure *structure;
+		GstEvent *event;
+		gint num = GPOINTER_TO_INT(user_data);
+
+		structure = gst_structure_new("dtmf-event",
+						"type", G_TYPE_INT, 1,
+						"number", G_TYPE_INT, num,
+						"start", G_TYPE_BOOLEAN, FALSE, NULL);
+
+		event = gst_event_new_custom(GST_EVENT_CUSTOM_UPSTREAM, structure);
+		gst_element_send_event(pipeline, event);
+	}
+}
+
+static inline GtkWidget *
+phone_create_button(const gchar *text_hi, const gchar *text_lo)
+{
+	GtkWidget *button;
+	GtkWidget *label_hi;
+	GtkWidget *label_lo;
+	GtkWidget *grid;
+
+	grid = gtk_vbox_new(TRUE, 0);
+
+	button = gtk_button_new();
+	label_hi = gtk_label_new(text_hi);
+	gtk_misc_set_alignment(GTK_MISC(label_hi), 0.5, 0.5);
+	gtk_box_pack_end(GTK_BOX(grid), label_hi, FALSE, TRUE, 0);
+	label_lo = gtk_label_new(text_lo);
+	gtk_misc_set_alignment(GTK_MISC(label_lo), 0.5, 0.5);
+	gtk_label_set_use_markup(GTK_LABEL(label_lo), TRUE);
+	gtk_box_pack_end(GTK_BOX(grid), label_lo, FALSE, TRUE, 0);
+	gtk_container_add(GTK_CONTAINER(button), grid);
+
+	return button;
+}
+
+static GtkWidget *
+pidgin_media_add_dtmf_widget(PidginMedia *gtkmedia,
+		PurpleMediaSessionType type, const gchar *sid)
+{
+	GtkWidget *grid = gtk_table_new(4, 3, TRUE);
+	GtkWidget *button1;
+	GtkWidget *button2;
+	GtkWidget *button3;
+	GtkWidget *button4;
+	GtkWidget *button5;
+	GtkWidget *button6;
+	GtkWidget *button7;
+	GtkWidget *button8;
+	GtkWidget *button9;
+	GtkWidget *button_asterisk;
+	GtkWidget *button0;
+	GtkWidget *button_pound;
+
+	/* Button 1 */
+	button1 = phone_create_button("o_o", "<b>1</b>");
+	g_signal_connect(button1, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(1));
+	g_signal_connect(button1, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(1));
+	gtk_table_attach(GTK_TABLE(grid), button1, 0, 1, 0, 1, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 2 */
+	button2 = phone_create_button("ABC", "<b>2</b>");
+	g_signal_connect(button2, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(2));
+	g_signal_connect(button2, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(2));
+	gtk_table_attach(GTK_TABLE(grid), button2, 1, 2, 0, 1, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 3 */
+	button3 = phone_create_button("DEF", "<b>3</b>");
+	g_signal_connect(button3, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(3));
+	g_signal_connect(button3, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(3));
+	gtk_table_attach(GTK_TABLE(grid), button3, 2, 3, 0, 1, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 4 */
+	button4 = phone_create_button("GHI", "<b>4</b>");
+	g_signal_connect(button4, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(4));
+	g_signal_connect(button4, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(4));
+	gtk_table_attach(GTK_TABLE(grid), button4, 0, 1, 1, 2, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 5 */
+	button5 = phone_create_button("JKL", "<b>5</b>");
+	g_signal_connect(button5, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(5));
+	g_signal_connect(button5, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(5));
+	gtk_table_attach(GTK_TABLE(grid), button5, 1, 2, 1, 2, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 6 */
+	button6 = phone_create_button("MNO", "<b>6</b>");
+	g_signal_connect(button6, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(6));
+	g_signal_connect(button6, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(6));
+	gtk_table_attach(GTK_TABLE(grid), button6, 2, 3, 1, 2, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 7 */
+	button7 = phone_create_button("PQRS", "<b>7</b>");
+	g_signal_connect(button7, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(7));
+	g_signal_connect(button7, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(7));
+	gtk_table_attach(GTK_TABLE(grid), button7, 0, 1, 2, 3, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 8 */
+	button8 = phone_create_button("TUV", "<b>8</b>");
+	g_signal_connect(button8, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(8));
+	g_signal_connect(button8, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(8));
+	gtk_table_attach(GTK_TABLE(grid), button8, 1, 2, 2, 3, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 9 */
+	button9 = phone_create_button("WXYZ", "<b>9</b>");
+	g_signal_connect(button9, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(9));
+	g_signal_connect(button9, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(9));
+	gtk_table_attach(GTK_TABLE(grid), button9, 2, 3, 2, 3, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button Asterisk */
+	button_asterisk = phone_create_button(_("DEL"), "<b>*</b>");
+	g_signal_connect(button_asterisk, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(10));
+	g_signal_connect(button_asterisk, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(10));
+	gtk_table_attach(GTK_TABLE(grid), button_asterisk, 0, 1, 3, 4, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button 0 */
+	button0 = phone_create_button("", "<b>0</b>");
+	g_signal_connect(button0, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(0));
+	g_signal_connect(button0, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(0));
+	gtk_table_attach(GTK_TABLE(grid), button0, 1, 2, 3, 4, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	/* Button Pound */
+	button_pound = phone_create_button(_("RETURN"), "<b>#</b>");
+	g_signal_connect(button_pound, "pressed", G_CALLBACK(phone_dtmf_pressed_cb), GINT_TO_POINTER(11));
+	g_signal_connect(button_pound, "released", G_CALLBACK(phone_dtmf_released_cb), GINT_TO_POINTER(11));
+	gtk_table_attach(GTK_TABLE(grid), button_pound, 2, 3, 3, 4, GTK_FILL | GTK_EXPAND, GTK_FILL | GTK_EXPAND, 2, 2);
+
+	gtk_widget_show_all(grid);
+
+	return grid;
+}
+
+static void
 pidgin_media_ready_cb(PurpleMedia *media, PidginMedia *gtkmedia, const gchar *sid)
 {
 	GtkWidget *send_widget = NULL, *recv_widget = NULL, *button_widget = NULL;
@@ -889,6 +1054,10 @@
 		gtk_box_pack_end(GTK_BOX(recv_widget),
 				pidgin_media_add_audio_widget(gtkmedia,
 				PURPLE_MEDIA_SEND_AUDIO, NULL), FALSE, FALSE, 0);
+
+		gtk_box_pack_end(GTK_BOX(recv_widget),
+				pidgin_media_add_dtmf_widget(gtkmedia,
+				PURPLE_MEDIA_SEND_AUDIO, NULL), FALSE, FALSE, 0);
 	}
 
 	if (type & PURPLE_MEDIA_AUDIO &&
