Description: Send message (webview) in buddy pounce is unusable
Author: Ritesh Khadgaray <ritesh@canonical.com>
Forwarded: https://developer.pidgin.im/ticket/15400

Index: b/pidgin/gtkpounce.c
===================================================================
--- a/pidgin/gtkpounce.c
+++ b/pidgin/gtkpounce.c
@@ -687,6 +687,7 @@
 		= gtk_check_button_new_with_mnemonic(_("P_lay a sound"));
 
 	send_msg_imhtml = pidgin_create_imhtml(TRUE, &dialog->send_msg_entry, NULL, NULL);
+	gtk_widget_set_size_request (send_msg_imhtml, -1, 150);
 	reset_send_msg_entry(dialog, NULL);
 	dialog->exec_cmd_entry    = gtk_entry_new();
 	dialog->popup_entry       = gtk_entry_new();
