--- a/configure.ac
+++ b/configure.ac
@@ -127,6 +127,10 @@
 AX_REQUIRE_LIB([SDL_net], [SDL_NET],
                [SDL_net.h], [SDL_net], [SDLNet_Init])
 
+dnl Check for libexpat.
+AX_REQUIRE_LIB([expat], [EXPAT],
+               [expat.h], [expat], [XML_ParserCreate])
+
 dnl Check for zlib.
 AX_REQUIRE_LIB([zlib], [ZLIB],
                [zlib.h], [z], [zlibVersion])
@@ -250,7 +254,6 @@
 AlephOne.spec
 Source_Files/Makefile
 Source_Files/CSeries/Makefile
-Source_Files/Expat/Makefile
 Source_Files/FFmpeg/Makefile
 Source_Files/Files/Makefile
 Source_Files/GameWorld/Makefile
--- a/Source_Files/Makefile.am
+++ b/Source_Files/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in 
 
-SUBDIRS = CSeries Expat Files FFmpeg GameWorld Input LibNAT Lua Misc	\
+SUBDIRS = CSeries Files FFmpeg GameWorld Input LibNAT Lua Misc	\
    ModelView Network RenderMain RenderOther Sound TCPMess XML
 
 if MAKE_WINDOWS
@@ -14,13 +14,13 @@
   shell.cpp shell_misc.cpp
 
 alephone_LDADD = \
-  CSeries/libcseries.a Expat/libexpat.a Files/libfiles.a FFmpeg/libffmpeg.a \
+  CSeries/libcseries.a Files/libfiles.a FFmpeg/libffmpeg.a \
   GameWorld/libgameworld.a Input/libinput.a Lua/liba1lua.a Misc/libmisc.a \
   ModelView/libmodelview.a Network/libnetwork.a Network/Metaserver/libmetaserver.a \
   RenderMain/librendermain.a RenderOther/librenderother.a Sound/libsound.a \
   XML/libxml.a \
   \
-  CSeries/libcseries.a Expat/libexpat.a Files/libfiles.a FFmpeg/libffmpeg.a \
+  CSeries/libcseries.a Files/libfiles.a FFmpeg/libffmpeg.a \
   GameWorld/libgameworld.a LibNAT/libnat.a Input/libinput.a Lua/liba1lua.a Misc/libmisc.a \
   ModelView/libmodelview.a Network/libnetwork.a Network/Metaserver/libmetaserver.a \
   RenderMain/librendermain.a RenderOther/librenderother.a Sound/libsound.a \
@@ -33,7 +33,7 @@
   -I$(top_srcdir)/Source_Files/Network/Metaserver \
   -I$(top_srcdir)/Source_Files/FFmpeg -I$(top_srcdir)/Source_Files/RenderMain \
   -I$(top_srcdir)/Source_Files/RenderOther -I$(top_srcdir)/Source_Files/Sound \
-  -I$(top_srcdir)/Source_Files/XML -I$(top_srcdir)/Source_Files/Expat \
+  -I$(top_srcdir)/Source_Files/XML \
   -I$(top_srcdir)/Source_Files/TCPMess
 
 AlephOne_LDADD = $(alephone_LDADD) alephone-resources.o
--- a/Source_Files/LibNAT/Makefile.am
+++ b/Source_Files/LibNAT/Makefile.am
@@ -11,5 +11,3 @@
 
 EXTRA_libnat_a_SOURCES = os_win.c os_unix.c
 
-AM_CPPFLAGS = -I$(top_srcdir)/Source_Files/Expat
-
--- a/Source_Files/Misc/Makefile.am
+++ b/Source_Files/Misc/Makefile.am
@@ -35,5 +35,5 @@
   -I$(top_srcdir)/Source_Files/FFmpeg -I$(top_srcdir)/Source_Files/RenderMain \
   -I$(top_srcdir)/Source_Files/RenderOther -I$(top_srcdir)/Source_Files/Sound \
   -I$(top_srcdir)/Source_Files/TCPMess \
-  -I$(top_srcdir)/Source_Files/XML -I$(top_srcdir)/Source_Files/Expat \
+  -I$(top_srcdir)/Source_Files/XML \
   -I$(top_srcdir)/Source_Files
--- a/Source_Files/XML/Makefile.am
+++ b/Source_Files/XML/Makefile.am
@@ -20,4 +20,4 @@
   -I$(top_srcdir)/Source_Files/TCPMess \
   -I$(top_srcdir)/Source_Files/RenderMain \
   -I$(top_srcdir)/Source_Files/RenderOther -I$(top_srcdir)/Source_Files/Sound \
-  -I$(top_srcdir)/Source_Files/Expat -I$(top_srcdir)/Source_Files
+  -I$(top_srcdir)/Source_Files
